### ISSUE TITLE
Gradients are duplicated if declarations are on a new line

### ISSUE THREAD

**Issue by ocean90** [postcss/autoprefixer#527]
If you have something like this…

```
.newline {
    background-image:
        linear-gradient( 45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4 ),
        linear-gradient( 45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4 );
}
```

… then I expect to get something like this…

```
.newline {
    background-image:
        -webkit-linear-gradient(45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4 ),
        -webkit-linear-gradient( 45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4 );
    background-image:
        -o-linear-gradient( 45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4 ),
        -o-linear-gradient( 45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4 );
    background-image:
        linear-gradient( 45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4 ),
        linear-gradient( 45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4 );
}
```

… but you currently get this…

```
.newline {
    background-image:
        linear-gradient(45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4 ), linear-gradient( 45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4);
    background-image:
        -webkit-linear-gradient(45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4), -webkit-linear-gradient(45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4);
    background-image:
        -o-linear-gradient(45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4), -o-linear-gradient(45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4);
    background-image:
        linear-gradient( 45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4 ),
        linear-gradient( 45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4 );
}
```

The newlines are removed and you get a duplicate rule for `linear-gradient`.

**Comment by ben-eb** [comment]
Thanks for reporting. Is it the addition of newlines that is causing this?

**Comment by ocean90** [comment]
@ben-eb Yes. For example
```
.newline {
    background-image: linear-gradient(45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4), linear-gradient(45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4);
}
```

or

```
.newline {
    background-image:
        linear-gradient(45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4), linear-gradient(45deg, #c4c4c4 25%, transparent 25%, transparent 75%, #c4c4c4 75%, #c4c4c4);
}
```

works as expected.

**Comment by ben-eb** [comment]
I'm not familiar enough with the code to do this, @TrySound will have a look into it. :smile:

**Comment by ai** [comment]
Maybe somebody want to rewrite gradients hack on top of @trysound value parser?

Anyway, I do not think that this task is so important. Somebody should fix it.

**Comment by TrySound** [comment]
@ai Okay

**Comment by ai** [comment]
Done 416904d

**Comment by ai** [comment]
Released in 6.1
